packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt

```diff
@@ -32,174 +21,177 @@ import expo.modules.imagepicker.fileproviders.CacheFileProvider
     }
 
-    mPermissions.askForPermissions(permissionsResponseHandler, Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.CAMERA)
-  }
+    OnActivityDestroys {
+      mWasHostDestroyedWhileCropping = true
+    }
+
+    OnActivityEntersForeground {
+      mWasHostDestroyedWhileCropping = false
+    }
 
-  // NOTE: Currently not reentrant / doesn't support concurrent requests
-  @ExpoMethod
-  fun launchImageLibraryAsync(options: Map<String, Any?>, promise: Promise) {
-    val pickerOptions = optionsFromMap(options, promise) ?: return
-
-    val libraryIntent = Intent().apply {
-      when (pickerOptions.mediaTypes) {
-        MediaTypes.IMAGES -> type = "image/*"
-        MediaTypes.VIDEOS -> type = "video/*"
-        MediaTypes.ALL -> {
-          type = "*/*"
-          putExtra(Intent.EXTRA_MIME_TYPES, arrayOf("image/*", "video/*"))
+    OnActivityResult { activity, (requestCode, resultCode, data) ->
+      if (shouldHandleOnActivityResult(activity, requestCode)) {
+        var pickerOptions = mPickerOptions!!
+        val promise = if (mWasHostDestroyedWhileCropping && mPromise !is PendingPromise) {
+          if (pickerOptions.base64) {
+            // we know that the activity was killed and we don't want to store
+            // base64 into `SharedPreferences`...
+            pickerOptions = ImagePickerOptions().apply {
+              quality = pickerOptions.quality
+              allowsEditing = pickerOptions.allowsEditing
+              aspect = pickerOptions.aspect
+              base64 = false
+              mediaTypes = pickerOptions.mediaTypes
+              exif = pickerOptions.exif
+              videoMaxDuration = pickerOptions.videoMaxDuration
+            }
+            // ...but we need to remember to add it later.
+            PendingPromise(pickerResultStore, isBase64 = true)
+          } else {
+            PendingPromise(pickerResultStore)
+          }
+        } else {
+          mPromise!!
         }
-      }
 
-      action = Intent.ACTION_GET_CONTENT
+        mPromise = null
+        mPickerOptions = null
+
+        handleOnActivityResult(promise, activity, requestCode, resultCode, data, pickerOptions)
+      }
     }
 
-    startActivityOnResult(libraryIntent, ImagePickerConstants.REQUEST_LAUNCH_IMAGE_LIBRARY, promise, pickerOptions)
+    // endregion
   }
 
-  //endregion
+  // endregion
 
-  //region helpers
+  // region helpers
 
-  private fun getMediaLibraryPermissions(writeOnly: Boolean): Array<String> {
-    return if (writeOnly) {
-      arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE)
-    } else {
-      arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.READ_EXTERNAL_STORAGE)
-    }
-  }
+  private fun getMediaLibraryPermissions(writeOnly: Boolean): Array<String> =
+    if (writeOnly) arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE)
+    else arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.READ_EXTERNAL_STORAGE)
```

Comment:
Afaik we skip braces only in one-liners
```suggestion
    if (writeOnly) { 
      arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE)
    } else { 
      arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.READ_EXTERNAL_STORAGE)
    }
```